pkgs/applications/editors/vscode/generic.nix

```diff
@@ -92,6 +93,8 @@ in
         # Override the previously determined VSCODE_PATH with the one we know to be correct
         sed -i "/ELECTRON=/iVSCODE_PATH='$out/lib/vscode'" $out/bin/${executableName}
         grep -q "VSCODE_PATH='$out/lib/vscode'" $out/bin/${executableName} # check if sed succeeded
+
+        ${postInstall}
```

Comment:
Instead of putting this in `installPhase`, put it as an attribute within the derivation while replacing `${postInstall}` with `runHook postInstall`.

```
stdenv.mkDerivation {
  ...
  inherit postInstall;
  ...
  installPhase = ''
    ...
    runHook postInstall
  '';
}
```